frontends/web/src/routes/bitsurance/account.tsx

```diff
@@ -48,17 +50,22 @@ export const BitsuranceAccount = ({ code, accounts }: TProps) => {
       alertUser(response.errorMessage);
       return;
     }
-    const options = accounts
-      // btc accounts that have never been insured, or with a canceled
-      // insurance contract, can be used to make a new contract.
-      .filter(account => account.coinCode === 'btc' &&
-          (!account.bitsuranceStatus
-           || account.bitsuranceStatus === 'canceled'
-           || account.bitsuranceStatus === 'refused'))
-      .map(({ name, code, coinCode }) => (
-        { label: name, value: code, coinCode, disabled: false }
-      ));
-    setBtcAccounts(await setOptionBalances(options));
+
+    // btc accounts that have never been insured, or with a canceled
+    // insurance contract, can be used to make a new contract.
+    const btcAccountz = accounts.filter(account => account.coinCode === 'btc' &&
+        (!account.bitsuranceStatus
+         || account.bitsuranceStatus === 'canceled'
+        || account.bitsuranceStatus === 'refused'));
+
+    const groupedBTCAccounts = getAccountsByKeystore(btcAccountz);
+    const groupedOpts: TGroupedOption[] = groupedBTCAccounts.map(grouped => ({
+      label: `${grouped.keystore.name} ${isAmbiguiousName(grouped.keystore.name, groupedBTCAccounts) ? `(${grouped.keystore.rootFingerprint})` : ''}`,
+      connected: grouped.keystore.connected,
+      options: grouped.accounts.map((account) => ({ label: account.name, value: account.code, coinCode: account.coinCode, disabled: false })) as TOption[]
+    }));
```

Comment:
is this map callback function the same as in select-receive and buy-info, can you move it to a shared function?

Reply:
Yeah, it's related to [this](https://github.com/digitalbitbox/bitbox-wallet-app/pull/2497#discussion_r1461681393). I'll refactor.